compiler/qsc_linter/src/lints/hir.rs

```diff
@@ -12,7 +12,7 @@ use crate::linter::hir::declare_hir_lints;
 use super::lint;
 
 declare_hir_lints! {
-    (NeedlessOperation, LintLevel::Warn, "operation does not contain any quantum operations", "this callable can be declared as a function instead")
+    (NeedlessOperation, LintLevel::Allow, "operation does not contain any quantum operations", "this callable can be declared as a function instead")
```

Comment:
On the one hand, this seems like an elegant solution to our problem of wanting different behavior in the extension language service (and playground) vs QCOM. But on the other hand, it seems a little unfortunate that now someone needs to look in multiple places to see what the default level of a lint is (here and in the overrides passed by the callers). I'm not quite sure how to reconcile that in my head... we get what we want, but at the cost of a potential maintenance burden in the future whose size I'm unsure of.

Since QCOM still uses `checkCode` could we instead go the other way where this spot stays as a warning and `checkCode` uses the override to turn this (and perhaps other future lints) off for that scenario?

Reply:
QCOM doesn't use `checkCode` for squiggles (though you're right, it does use `checkCode` in some places, and I hadn't even considered that. But I think this change is fine in that regard).

I've resigned myself to just having overrides in multiple places. I don't think we can get around that. What you're suggesting would just shift the overrides to a different spot, but not get rid of them completely.

Mayyybe eventually we can simplify a little bit by having all _these_ hardcoded defaults be the same, e.g. make them all `Allow` (opt-in) or `Warn` (opt-out). Then, at least, when you look at an `update_configuration` call in the source, you can rest assured that's the definitive list for that scenario, and not something to be merged with hardcoded defaults elsewhere.

I can't do that right now though because I'm trying to avoid a QCOM change....